src/test/java/com/radixdlt/atommodel/validators/ValidatorConstraintScryptTest.java

```diff
@@ -0,0 +1,216 @@
+package com.radixdlt.atommodel.validators;
+
+import com.radixdlt.atommodel.validators.ValidatorConstraintScrypt.ValidatorTransitionProcedure;
+import com.radixdlt.atomos.CMAtomOS;
+import com.radixdlt.atomos.Result;
+import com.radixdlt.constraintmachine.Particle;
+import com.radixdlt.constraintmachine.Spin;
+import com.radixdlt.constraintmachine.WitnessData;
+import com.radixdlt.identifiers.RadixAddress;
+import com.radixdlt.store.CMStore;
+import com.radixdlt.store.CMStores;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Exercise the critical invariants of the {@link ValidatorConstraintScrypt}.
+ *
+ * The invariants tested here follow the flow of the validator FSM:

```

Comment:
Kudos for docs, but list needs to be formatted as Javadoc to be readable in Javadoc output.
Something like:
```
/**
 * Exercise the critical invariants of the {@link ValidatorConstraintScrypt}.
 *
 * The invariants tested here follow the flow of the validator FSM:
 * <ul>
 *   <li>unregistered validator with nonce 0 is virtualised as UP</li>
 *   <li>validator address must be specified</li>
 *   <li>validator transitions must be between validators of the same address</li>
 *   <li>the nonce must increase by exactly one with every transition</li>
 *   <li>validator transitions must be signed by the address</li>
 * </ul>
 *
 * These tests are implemented by either testing the static check directly or the generated transition procedure.
 */
```